ethereum/contracts/ERC20App.sol

```diff
@@ -52,19 +56,27 @@ contract ERC20App {
         return abi.encode(_sender, _recipient, _tokenAddr, _amount, _nonce);
     }
 
-    function submit(bytes memory data)
+    function unlockERC20(bytes memory _data)
```

Comment:
How do we handle auth for this function? Presently, anyone can call it, right?

Reply:
Auth should be restricted to the Broker contract, which checks the signature before calling these functions. A downside of this approach is that the Broker would need to be registered on the app as well.

Reply:
Okay, but the logic to restrict access to the broker is not here currently, is it? 

Could you add a TODO in code? Thanks!